src/codeflare_sdk/ray/cluster/generate_yaml.py

```diff
@@ -88,9 +94,17 @@ def update_names(
 
 def update_image(spec, image):
     containers = spec.get("containers")
-    if image != "":
-        for container in containers:
-            container["image"] = image
+    if not image:
+        python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
+        try:
+            if python_version in SUPPORTED_PYTHON_VERSIONS:
+                image = SUPPORTED_PYTHON_VERSIONS[python_version]
+        except Exception:  # pragma: no cover
+            print(
+                f"Python version '{python_version}' is not supported. Only {', '.join(SUPPORTED_PYTHON_VERSIONS.keys())} are supported."
```

Comment:
Lets be explicit about the issue here. i.e. "No default Ray image defined for {python version}. Please provide your own image or use one of the following python versions."